src/scmdata/run.py

```diff
@@ -597,7 +597,7 @@ def __setitem__(
 
         Notes
         -----
-        If the meta values changes are applied to a filtered subset, the change will be reflected
```

Comment:
What do we mean with the change here? I understand the current text to mean, "If you filter, then change meta, the original object will also change". Is that not the case?

Reply:
No, this test:

```python
def test_set_item_filter(scm_run):
    filtered = scm_run.filter(variable="Primary Energy|Coal")
    filtered["model"] = "other_model"
    assert (scm_run["model"] == ["a_iam", "other_model", "a_iam"]).all()
```

fails on master (and on this branch) with:
```
========================= 1 failed, 1 warning in 0.05s =========================
FAILED                         [100%]
tests/unit/test_item_filter.py:0 (test_set_item_filter)
0    a_iam
1    a_iam
2    a_iam
Name: model, dtype: object != ['a_iam', 'other_model', 'a_iam']
```

However, I have seen this behaviour live, where the filtered subset did change. I don't understand when that happens, so figured better tell people to not rely on this either way.

Reply:
Hmmm. IIRC the design goal was for filtered subsets to *not* update their "parent" dataset as that can be confusing and hard to guarantee. This isn't the most performant as it leads to multiple copies, but it is the easiest to understand. For the brave, there are `inplace` arguments sprinkled around the API to opt out of that behaviour. 

For modifying meta values inplace, we added `set_meta` as we often did a filter, update and append workflow. This function is a prime candidate for a performance review as it currently is rather naive.

 I wonder whether this documentation is simply out of date. 

If you have seen different behaviour in the wild that suggests this text is correct until we have a stronger guarantee. I'd be interested in seeing a test case if you see the different behaviour.

Reply:
Yes, I saw this behaviour in the wild in ndc-quantifications the other week, actually. Tests started passing when I added [this copy](https://gitlab.com/climate-resource/ndcs/ndc-quantification/-/blob/main/src/ndcs/variant/future_baseline_emissions_growth.py#L45). However, I was not able to craft a test case for it :shrug: .

Reply:
I added tests for it and removed the note. @lewisjared Are you happy with that? Did I miss an important test case for this?